AnkiDroid/src/main/java/com/ichi2/anki/FieldEditText.java

```diff
@@ -137,51 +142,48 @@ public void setImagePasteListener(ImagePasteListener imageListener) {
     }
 
     @Override
-    @SuppressWarnings("deprecation") // Tracked in #9775
+    @KotlinCleanup("add extension method to iterate clip items")
     public InputConnection onCreateInputConnection(EditorInfo editorInfo) {
         InputConnection inputConnection = super.onCreateInputConnection(editorInfo);
         if (inputConnection == null) {
             return null;
         }
         EditorInfoCompat.setContentMimeTypes(editorInfo, IMAGE_MIME_TYPES);
-        return InputConnectionCompat.createWrapper(inputConnection, editorInfo, (contentInfo, flags, opts) -> {
 
-            if (mImageListener == null) {
-                return false;
-            }
+        ViewCompat.setOnReceiveContentListener(this, IMAGE_MIME_TYPES, (view, payload) -> {
 
-            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N_MR1  && (flags &
-                    InputConnection.INPUT_CONTENT_GRANT_READ_URI_PERMISSION) != 0) {
-                try {
-                    contentInfo.requestPermission();
-                }
-                catch (Exception e) {
-                    return false;
-                }
+            Pair<ContentInfoCompat, ContentInfoCompat> split = payload.partition(
+                    item -> item.getUri() != null);
+            ContentInfoCompat uriContent = split.first;
+            ContentInfoCompat remaining = split.second;
+
+            if (mImageListener == null || uriContent == null) {
+                return remaining;
             }
 
-            ClipDescription description = contentInfo.getDescription();
+            ClipData clip = uriContent.getClip();
+            ClipDescription description = clip.getDescription();
 
             if (!ClipboardUtil.hasImage(description)) {
-                return false;
+                return remaining;
             }
 
-            try {
-                if (!onImagePaste(contentInfo.getContentUri())) {
-                    return false;
+            for (int i = 0; i < clip.getItemCount(); i++){
+                Uri uri = clip.getItemAt(i).getUri();
+                try {
+                    mImageListener.onImagePaste(this, uri);
+                }
+                catch (Exception e) {
```

Comment:
Our java style guide is not enforced by a lint check, this is just a spacing suggestion, it has nothing to do with functionality but matches java style here

```suggestion
            for (int i = 0; i < clip.getItemCount(); i++) {
                Uri uri = clip.getItemAt(i).getUri();
                try {
                    mImageListener.onImagePaste(this, uri);
                } catch (Exception e) {
```